rq/worker.py

```diff
@@ -698,6 +698,7 @@ def perform_job(self, job, queue):
         try:
             with self.death_penalty_class(job.timeout or self.queue_class.DEFAULT_TIMEOUT):
                 rv = job.perform()
+            job.refresh()
```

Comment:
The problem with moving the refresh call further in is that it would clobber any information that rq sets. From a high-level design point of view, it makes sense to do the save right before the job execution, and the fetch right afterwards. The state stored in redis should be considered canonical, and so when passing around the control of the state between the worker and the job, synchronization should occur at each transfer to prevent ever having an out-of-date state.

Another consideration which I am not qualified to assess. For those who subclass Worker, where should the refresh() call go in order to ensure that this behavior is maintained in the subclasses?